File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <string>

using namespace std;

typedef struct Stack {
	int data[10000];
	int size = 0;

	void push(int val)
	{
		data[size++] = val;
	}

	int pop()
	{
		if (size > 0) {
			return data[--size];
		}
		else {
			return -1;
		}	
	}

	int empty()
	{
		if (size == 0) return 1;
		else return 0;
	}

	int top()
	{
		if (empty()) return -1;
		else return data[size - 1];
	}

}Stack;



int main()
{
	Stack s;
	
	int n;
	cin >> n;

	while (n--) {
		string cmd;
		cin >> cmd;

		if (cmd == "push") {
			int num;
			cin >> num;
			s.push(num);
		}
		else if (cmd == "top") {
			cout << s.top() << '\n';
		}
		else if (cmd == "size") {
			cout << s.size << '\n';
		}
		else if (cmd == "empty") {
			cout << s.empty() << '\n';
		}
		else if (cmd == "pop") {
			cout << s.pop() << '\n';
		}
	}
	
	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		Deque<String> stack = new ArrayDeque<>();
		
		while(n-- > 0){
			StringTokenizer st = new StringTokenizer(br.readLine());
			String cmd = st.nextToken();
			if(cmd.equals("push")){
				String k = st.nextToken();
				stack.push(k);
			}else if(cmd.equals("pop")){
				if(stack.isEmpty()) {
					System.out.println(-1);
				}
				else {
					System.out.println(stack.peek());
					stack.pop();
				}
			}else if(cmd.equals("size")){
				System.out.println(stack.size());
			}else if(cmd.equals("empty")){
				if(stack.isEmpty()) System.out.println(1);
				else System.out.println(0);
			}else if(cmd.equals("top")){
				if(stack.isEmpty()) System.out.println(-1);
				else System.out.println(stack.peek());
			}
		}
		
	}
}